src/Microsoft.ML.Data/DataLoadSave/Text/TextSaver.cs

```diff
@@ -491,15 +491,7 @@ private TextLoader.Column GetColumn(string name, ColumnType type, int? start)
             ColumnType itemType = vectorType?.ItemType ?? type;
             if (itemType is KeyType key)
             {
-                if (!key.Contiguous)
-                    keyRange = new KeyRange(key.Min, contiguous: false);
-                else if (key.Count == 0)
-                    keyRange = new KeyRange(key.Min);
-                else
-                {
-                    Contracts.Assert(key.Count >= 1);
-                    keyRange = new KeyRange(key.Min, key.Min + (ulong)(key.Count - 1));
-                }
+                keyRange = new KeyRange(key.Count - 1);
```

Comment:
>KeyRange [](start = 31, length = 8)

Why do we still need `KeyRange`?

Reply:
`KeyRange` is used mainly in the arguments, for example in the TextLoader arguments. It takes care of parsing, and unparsing into the form U4[100] (and the other format that we discussed in the PR). It's therefore mainly a utility class for parsing and for now I think it makes sense to keep it.

---
In reply to: [249920350](https://github.com/dotnet/machinelearning/pull/2146#discussion_r249920350) [](ancestors = 249920350)